### PR TITLE
Fix rve issue and add rve tests

### PR DESCRIPTION
**Pull request by kmarchais**
This PR fixes a bug brought by the deprecation of Rve's `dim_x`, `dim_y` and `dim_z` attributes.

In addition to this fix, Deprecation Warnings for these parameters, and tests for the Rve class have been added.

### CODE REVIEW

File: microgen/rve.py [hunk 1]
```diff
@@ -22,45 +22,52 @@ class Rve:
...
         self,
         dim_x: Optional[float] = None,
         dim_y: Optional[float] = None,
         dim_z: Optional[float] = None,
         center: Union[np.ndarray, Tuple, List] = (0, 0, 0),
-        dim: Optional[Union[float, np.ndarray, Tuple, List]] = 1,
+        dim: Union[float, np.ndarray, Tuple, List] = 1,
     ) -> None:
-        if isinstance(center, (Tuple, List)) and len(center) == _DIM:
+        if isinstance(center, (tuple, list)) and len(center) == _DIM:
```
Comment (ricardo0115) [inline microgen/rve.py:33]: We are supposed to use `tuple` or `Tuple` ? to keep the compatibility with earlier python versions

File: microgen/rve.py [hunk 2]
```diff
@@ -22,45 +22,52 @@ class Rve:
...
+        if dim_x is not None and dim_y is not None and dim_z is not None:
+            self.dim = np.array([dim_x, dim_y, dim_z])
             warnings.warn(
-                "dim_x, dim_y, dim_z will be deprecated, use 'dim' instead.",
+                f"dim_x, dim_y, dim_z are deprecated, use 'dim' instead. \
+                    Now dim is set to [{self.dim[0]}, {self.dim[1]}, {self.dim[2]}]",
                 DeprecationWarning,
+                stacklevel=2,
+            )
+        elif dim_x is not None or dim_y is not None or dim_z is not None:
```
Comment (ricardo0115) [inline microgen/rve.py:57]: In which case we are in this condition ?

File: microgen/rve.py (hunk 1 above)
Comment (kmarchais) [reply microgen/rve.py:33]: This is generally a problem with python 3.8, but mypy was showing a problem here and requesting this change. Tested with python 3.8 and it works fine.

File: microgen/rve.py (hunk 2 above)
Comment (kmarchais) [reply microgen/rve.py:57]: In case of `Rve(dim_x=2)` to get `dim == [2, 1, 1]` as it would be in microgen 1.0.0

Comment (kmarchais) [reply microgen/rve.py]: Refactored this part to take care of both situations in the same if

**Pull request merged**